### DISCUSSION TITLE
DDR AI Movement block

### DISCUSSION THREAD

**Discussion by FrozenEmpire** [ChilliHugger/The-Lords-Of-Midnight#198] [Code Algorithm Questions] [answered]
Hi Chris,

In the DDR GetLocInfo() routine, there is a check for AI characters as below.
I don't think it has any impact, since AI characters move using Cmd_WalkForward, which does not check the lif_moveforward flag

// we shouldn't leave a location that has enemies in
`
if( IsAIControlled() ) {
    if( info->foe.armies || info->foe.characters ) {
        info->flags.Reset(lif_moveforward);
    }
}
`

Cheers,
Andrew

**Comment by IcemarkUK** [comment, marked as answer]
Check out here...
https://github.com/ChilliHugger/The-Lords-Of-Midnight/blob/f0242bb21c4e99e8afa61f33cf421807c217fb95/main/midnight/src/tme/scenarios/ddr/ddr_character.cpp#L1270-L1280

**Comment by FrozenEmpire** [comment]
Dang! Missed that :)
